tests/test-extensions/transactional-extension/src/main/java/org/mule/test/transactional/TransactionalOperations.java

```diff
@@ -31,6 +35,12 @@ public void verifyTransactionRolledback(@Connection TestTransactionalConnection
     checkState(connection.isTransactionRolledback(), "transaction not rolled back");
   }
 
+  public void verifySameConnection(@Connection TestTransactionalConnection connection,
+                                   @Content TestTransactionalConnection transactionalMessage) {
+    checkState(Objects.equals(connection.getConnectionId(), transactionalMessage.getConnectionId()),
```

Comment:
if you're asserting for the same, then use `==`. Don't speculate on the equals() implementation